File: microsoft/3/3.cpp
#include <bits/stdc++.h> 
using namespace std;  
  
int subArraySum(int arr[], int n, int sum)  
{  
    int curr_sum, i, j;  
    for (i = 0; i < n; i++)  
    {  
        curr_sum = arr[i];  
          for (j = i + 1; j <= n; j++)  
        {  
            if (curr_sum == sum)  
            {  
                cout << "Sum found between indexes " 
                     << i << " and " << j - 1;  
                return 1;  
            }  
            if (curr_sum > sum || j == n)  
                break;  
        curr_sum = curr_sum + arr[j];  
        }  
    }  
  
    cout << "No subarray found";  
    return 0;  
}  
int main()  
{  
    int n;
    cin>>n;
    int a[n];
    for(int i=0;i<n;i++){
    	cin>>a[i];
	}
	n = sizeof(a) / sizeof(a[0]);  
    int sum;
	cin>>sum; 
    subArraySum(a, n, sum);  
    return 0;  
}  
  

File: Microsoft/3/3.py
def subArraySum(arr, n, sum):
    for i in range(n):
        curr_sum = arr[i]
        j = i + 1
        while j <= n:

            if curr_sum == sum:
                print("Sum found between")
                print("indexes %d and %d" % (i, j - 1))

                return 1

            if curr_sum > sum or j == n:
                break

            curr_sum = curr_sum + arr[j]
            j += 1

    print("No subarray found")
    return 0

arr = []
n = int(input());
arr = [int(x) for x in input().split()]
n = len(arr)
sum =int(input())
subArraySum(arr, n, sum)
